lib/Service/Social/MastodonProvider.php

```diff
@@ -79,9 +79,9 @@ public function getImageUrls(array $contact):array {
 	 */
 	public function getImageUrl(string $profileUrl):?string {
 		try {
-			$result = $this->httpClient->get($profileUrl);
+			$result = $this->httpClient->get($profileUrl, ['headers' => ['Accept' => 'application/json']]);
 			$jsonResult = json_decode($result->getBody());
-			return $jsonResult->avatar;
+			return $jsonResult->icon->url;
```

Comment:
nit: deserialize into an associative array and use the null coalesce operator to not cause errors with missing keys

Reply:
good comment. I modfied it as follows:
https://github.com/nextcloud/contacts/blob/068c135dcc2017531b906cb3794096b189045469/lib/Service/Social/MastodonProvider.php#L83